L1Trigger/L1THGCalUtilities/python/hgcalTriggerChains.py

```diff
@@ -90,14 +118,80 @@ def create_sequences(self, process):
+                selector_sequence *= getattr(process, selector_name)
+            if ntuple!='' and not hasattr(process, ntuple_name):
+                setattr(process, ntuple_name, self.ntuple[ntuple](process, ntuple_inputs))
+                ntuple_sequence *= getattr(process, ntuple_name)
+        for vfe,calo,backend1,backend2,selector,ntuple in self.truth_be_chain:
+            calo_name = '{0}{1}'.format(vfe, calo)
+            backend1_name = '{0}{1}{2}'.format(vfe, calo, backend1)
+            backend2_name = '{0}{1}{2}{3}'.format(vfe, calo, backend1, backend2)
+            selector_name = '{0}{1}{2}{3}{4}'.format(vfe, calo, backend1, backend2, selector)
+            ntuple_name = '{0}{1}{2}{3}{4}{5}'.format(vfe, calo, backend1, backend2, selector, ntuple)
+            if selector=='':
+                ntuple_inputs = [
+                        calo_name,
+                        backend1_name+':HGCalBackendLayer1Processor2DClustering',
+                        backend2_name+':HGCalBackendLayer2Processor3DClustering'
+                        ]
+            else:
+                ntuple_inputs = [
+                        calo_name,
+                        backend1_name+':HGCalBackendLayer1Processor2DClustering',
+                        selector_name]
+            if not hasattr(process, vfe):
+                setattr(process, vfe, self.vfe[vfe](process))
+                vfe_task.add(getattr(process, vfe))
+            if not hasattr(process, calo_name):
+                setattr(process, calo_name, self.calo[calo](process, vfe))
+                calo_task.add(getattr(process, calo_name))
+            if not hasattr(process, backend1_name):
+                setattr(process, backend1_name, self.backend1[backend1](process, calo_name))
+                backend1_task.add(getattr(process, backend1_name))
+            if not hasattr(process, backend2_name):
+                setattr(process, backend2_name, self.backend2[backend2](process, backend1_name))
+                backend2_task.add(getattr(process, backend2_name))
+            if selector!='' and not hasattr(process, selector_name):
+                setattr(process, selector_name, self.selector[selector](process, backend2_name))
+                selector_sequence *= getattr(process, selector_name)
+            if ntuple!='' and not hasattr(process, ntuple_name):
+                setattr(process, ntuple_name, self.ntuple[ntuple](process, ntuple_inputs))
+                ntuple_sequence *= getattr(process, ntuple_name)
+
         vfe_task.remove(tmp)
         concentrator_task.remove(tmp)
+        calo_task.remove(tmp)
         backend1_task.remove(tmp)
         backend2_task.remove(tmp)
         selector_sequence.remove(tmpseq)
         ntuple_sequence.remove(tmpseq)
         process.globalReplace('hgcalVFE', vfe_task)
         process.globalReplace('hgcalConcentrator', concentrator_task)
+        process.globalReplace('truthCellsProd', calo_task)
```

Comment:
Instead of letting the user redefine the Task in the cfg file to make this globalReplace work, I would do it somewhere before in `create_sequence`, such that it is transparent for the user.

Reply:
I moved all the caloTruthProducer setup to hgcalTriggerChains.py. I'm not sure if this is what you meant, or if there's a better way. So let me know what you think.